lib/Dialect/ESI/ESIPasses.cpp

```diff
@@ -180,6 +193,48 @@ circt::rtl::RTLExternModuleOp ESIRTLBuilder::declareStage() {
   return declaredStage;
 }
 
+/// Return the InterfaceType which corresponds to an ESI port type. If it
+/// doesn't exist in the cache, build the InterfaceOp and the corresponding
+/// type.
+InterfaceOp ESIRTLBuilder::getOrConstructInterface(ChannelPort t) {
+  auto ifaceIter = portTypeLookup.find(t);
```

Comment:
Rather than caching the interfaces in `portTypeLookup`, couldn't we just use the IR's symbol table? I'm imagining something like this:
1. call a (hypothetical) helper `name = getInterfaceName(t)` to get the symbolic name for an interface of channel type `t` (see previous comment about wanting to have unique names by channel type)
2. do a `module.lookup<InterfaceOp>(name)` to see if the interface exists
3. if not, create the interface using a modified `constructInterface`

Reply:
The problem is that said generated name isn't guaranteed to be unique in the symbol table, so I'd have to keep a lookup of ChannelPort to symbol name. As long as I'm storing a lookup table I might as well just keep the InterfaceOp reference.

Reply:
Hmm fair enough, I wasn't considering the name colliding with something else in the symbol table. The current implementation looks good to me.